lib/src/command/add.dart

```diff
@@ -94,20 +76,56 @@ class AddCommand extends PubCommand {
         help: 'Build executables in immediate dependencies.');
     argParser.addOption('directory',
         abbr: 'C', help: 'Run this in the directory <dir>.', valueHelp: 'dir');
+    argParser.addFlag(
+      'example',
+      help:
+          'Also update dependencies in `example/` after modifying pubspec.yaml in the root package (if it exists).',
+      hide: true,
+    );
   }
 
+  /// Parses the arguments that are allowed after the first "--".
+  late final followingPartsArgParser = () {
+    final argParser = ArgParser(
+        allowTrailingOptions: allowTrailingOptions,
+        usageLineLength: lineLength);
+    argParser.addFlag('dev',
+        abbr: 'd',
+        negatable: false,
+        help: 'Adds to the development dependencies instead.');
+
+    argParser.addOption('git-url', help: 'Git URL of the package');
+    argParser.addOption('git-ref',
+        help: 'Git branch or commit to be retrieved');
+    argParser.addOption('git-path', help: 'Path of git package in repository');
+    argParser.addOption('hosted-url', help: 'URL of package host server');
+    argParser.addOption('path', help: 'Add package from local path');
+    argParser.addOption(
+      'sdk',
+      help: 'add package from SDK source',
+      allowed: ['flutter'],
+      valueHelp: '[flutter]',
+    );
```

Comment:
Should we make all of these options hidden and deprecated?

Reply:
Done